File: python/P1512_Number_Good_Pairs.py
from typing import List

class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        dictionary = {}
        goodPairs = 0
        
        for num in nums:
            if(num in dictionary):
                count = dictionary[num]
                goodPairs += count
                dictionary[num] += num
            else:
                dictionary[num] = 1
        
        return goodPairs
        

solution = Solution()
nums = [1,1,1,1]
print(solution.numIdenticalPairs(nums))

File: java/src/main/java/P1512_Number_Good_Pairs.java
import java.util.HashMap;
import java.util.Map;

public class P1512_Number_Good_Pairs {

    public static void main(String args[]) {

        int[] nums = { 1, 2, 3, 1, 1, 3 };
        P1512_Number_Good_Pairs instance = new P1512_Number_Good_Pairs();
        int result = instance.numIdenticalPairs(nums);

        System.out.println(result);
    }

    public int numIdenticalPairs(int[] nums) {
        Map<Integer, Integer> frequencyMap = new HashMap<>();
        int goodPairs = 0;

        for (int num : nums) {
            if (frequencyMap.containsKey(num)) {
                int count = frequencyMap.get(num);
                goodPairs += count; 
                frequencyMap.put(num, count + 1);
            } else {
                frequencyMap.put(num, 1); 
            }
        }

        return goodPairs;
    }
}